src/NodeAnalyzer/ModelAnalyzer.php

```diff
@@ -52,26 +73,128 @@ public function getPrimaryKey(string $class): ?string
     }
 
     /**
+     * @param  class-string<Model>|ObjectType  $model
+     */
+    public function isQueryScopeOnModel(string|ObjectType $model, string $scopeName, Scope $scope): bool
+    {
+        if (! is_string($model)) {
+            /** @var class-string<Model> $model */
+            $model = $model->getClassName();
+        }
+
+        $classReflection = $this->getClass($model);
+
+        if ($classReflection->hasMethod('scope' . ucfirst($scopeName))) {
+            return true;
+        }
+
+        try {
+            $method = $classReflection->getMethod($scopeName, $scope);
+
+            if ($this->usesScopeAttribute($method)) {
+                return true;
+            }
+        } catch (MissingMethodFromReflectionException) {
+        }
+
+        return false;
+    }
```

Comment:
nit: you can remove the need for a try/catch by checking if the class has the method first:


```suggestion
        if (! $classReflection->hasMethod($scopeName)) {
            return false;
        }

        $method = $classReflection->getMethod($scopeName, $scope);

        return $this->usesScopeAttribute($method);
```